### PR TITLE
Add wopi documentation

### PR DESCRIPTION
**Pull request by mmattel**
Fixes: #272

This PR:

* adds an overview description page about the integration of WOPI and ocis which is necessary to understand the required settings in the example to be set and
* adapts the existing office example page accordingly.

Note the images have been created based on a very intense discussion and cooperation with @wkloucek (the base of the images were joint effort). We both have agreed that the images are needed to understand the process.

A language review is very welcomed.

The rendered version is currently visible on staging.

### CODE REVIEW

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (micbar) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:94]: I would change that. The referenced example files will get regular updates. I would prefer "the CS3Org WOPI server will be started with the version from the examples, which might not be the latest stable released version.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (micbar) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:109]: is the dot a typo?

Comment (micbar) [reply modules/ROOT/pages/conf-examples/office-integration.adoc:109]: what is `list_file_2.adoc`?

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (micbar) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:124]: maybe a hint would be good to run `docker-compose ps`

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (micbar) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:34]: This is not correct. The mimetypes are registered in the app provider by calling the discovery endpoint of the office application. The app registry is only needed to set the default app.

**Review by micbar — CHANGES_REQUESTED** [review]
See comments.

I would also like to have a second opinion on the detail level and the images.

**Comment by micbar** [comment]
@TheRealBro @xoxys @rhafer Can you take a look?

I am unsure if the detail level of this documentation is the right one for the target audience.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:24]:
```suggestion
* and Collabora and OnlyOffice containers.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:26]:
```suggestion
Note that it is not mandatory to use and configure both example Office Application stacks. Configure only the application stacks that you need.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:75]:
```suggestion
. firstly, the xref:conf-examples/ext-files/list_file_1.adoc[environment definition file,window=_blank] needs to be adapted. This file defines many basic settings in one location. Open it in a text editor and adjust it according to your needs.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:98]:
```suggestion
* You also must override the default WOPI JWT secret for a secure setup. Do this by setting `WOPI_JWT_SECRET` to a long and random string.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:100]:
```suggestion
* For OnlyOffice, wopi is not enabled by default. It has to be set explicitly via `WOPI_ENABLED: "true"`.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:109]:
```suggestion
. Though no changes are needed, adapt the xref:conf-examples/ext-files/list_file_2.adoc[UI representation,window=_blank] of the office applications for use in ownCloud Web if required. As an example, the settings currently do not have any icons defined that show up in the webUI with the names of available office applications to open. This can be changed manually.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (phil-davis) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:124]:
```suggestion
. Enter your Infinite Scale domain in a browser and you're ready to log in and open office documents in the web client. It might take a few minutes for all services to become operational because the containers need to download first. Just keep reloading the pages from time to time.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (phil-davis) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:9]:
```suggestion
:description: Infinite Scale uses the WOPI protocol to integrate office applications. This document describes the general flow and basic settings used.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (phil-davis) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:42]:
```suggestion
This image gives a brief overview of which components and settings are involved. It also shows where parts of an URL are defined, how an URL is assembled, and which component is using which part of the URL.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (phil-davis) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:48]:
```suggestion
With this image, administrators can get a more detailed view about the components involved down to a service level. It also includes steps triggered by the user and the requests and responses on a generic level. It shows how a reverse proxy fits into the picture. The reverse proxy is exemplary per level, where in reality, it usually is a redundant proxy environment configured per level. You can use any proxy of your choice like {traefik_url}[Traefik], {nginx_url}[nginx] or {apache_url}[Apache].
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (phil-davis) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:38]:
```suggestion
The following two sections describe the wopi integration in an overview and a more detailed image to get a better understanding of which components are involved and which basic settings need to be set. You can also extrapolate how to scale office application servers and how to use various office applications in parallel. Note that only a few basic configuration options are shown in the images below to support the process flow.
```

**Review by phil-davis — COMMENTED** [review]
Just some English suggestions. I am not totally familiar with all the technical content, so that needs review by others.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (EParzefall) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:26]:
```suggestion
Note that it is not mandatory to use and configure both example office application stacks. Configure only the application stacks that you need.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (EParzefall) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:34]:
```suggestion
* ocis.* for serving Infinite Scale
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (EParzefall) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:75]:
```suggestion
. Firstly, the xref:conf-examples/ext-files/list_file_1.adoc[environment definition file,window=_blank] needs to be adapted. This file defines many basic settings in one location. Open it in a text editor and adjust it according to your needs.
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (EParzefall) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:94]:
```suggestion
* By default, the CS3Org WOPI server will be started with the version from the examples, which might not be the latest stable released version. If you want to start a different or more recent version, you can set the `WOPISERVER_DOCKER_TAG=`. Available versions can be found on {wopiserver_url}[Docker Hub].
```

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (EParzefall) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:100]:
```suggestion
* For OnlyOffice, WOPI is not enabled by default. It has to be set explicitly via `WOPI_ENABLED: "true"`.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:1]:
```suggestion
= Office Applications Using WOPI
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:34]:
```suggestion
* configured for use with the WebUI via the xref:{s-path}/app-registry.adoc[App-Registry Service]. Here, the default office app is set. For more information see the xref:#wopi-coniguration-examples[app-registry.yaml] file.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:38]:
```suggestion
The following two sections describe the WOPI integration in an overview and a more detailed image to get a better understanding of which components are involved and which basic settings need to be set. You can also extrapolate how to scale office application servers and how to use various office applications in parallel. Note that only a few basic configuration options are shown in the images below to support the process flow.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:46]:
```suggestion
=== WOPI in Detail
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:54]:
```suggestion
* See below for an example WOPI configuration using Docker Compose, to get a first impression of how the WOPI configuration can be achieved. Both, _ocis environment variables_ and _ocis service configuration_ yaml files are used. See the xref:conf-examples/office-integration.adoc[Office Integration] document for an in depth explanation.
```

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (EParzefall) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:86]:
```suggestion
Note that github will not let you download a single directory easily. You can get the examples using the following methods requiring minimum space. Replace the version accordingly:
```

**Review by EParzefall — COMMENTED** [review]
language review

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (dragotin) [inline modules/ROOT/pages/conf-examples/office-integration.adoc:87]: Please remove the part about the Traefik Dashboard. Not applicable for this documentation.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (dragotin) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:18]: What about MS Office 365?

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (dragotin) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:34]: I think that is way to verbose. The App Provider should/will be handled later with documentation about extensions I would say. Can you shorten this?

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (dragotin) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:46]: Again, to verbose. Please keep the information about the WOPI protocol here limited, as that is not the scope of this document. 
Rather put that into the dev docs.

**Review by dragotin — CHANGES_REQUESTED** [review]
While I appreciate the work on for example the WOPI diagrams, I think that is much to deep for the admin docs. 

Please see comments inline.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc]: The dot ends the sentence and is part of the list-item.
The xref `list_file_x.adoc` is a clickable link where its content is opened in another window when clicked. This way we only have a link and not the whole content in the same page which would make it very hard to read.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc:87]: We are not the master/creator of the compose file which is referenced by this text. The compose file was created by Willy. We take what he provides. This is now "locked" in the v2.0.0. branch. If a change is required, one with the right knowledge can do it in the master branch of the ocis repo. When the next release like v2.1 is published, we can then change the description accordingly. You can see the whole picture more easily when looking into staging where you can see a rendered build...

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc:18]: I have been told by Willy, as far I remember, that only the above are supported. And I know from the past, that MS Office 365 support is "special". Lets have a chat/video about that. Note that we can add the line item with MS Office at any time afterwards.

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc]: Fixed by @EParzefall

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc]: Added with the next push

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: > The app registry is only needed to set the default app

This is not correct. `app-registry` defines which apps are available for which mimetypes, which icons they show, which names they get, which app is default when not overrided by the user etc and all that info is shown as list items in the webUI for the user whe he selects to open a file. See the example below.

```
  - mime_type: application/vnd.oasis.opendocument.text
    extension: odt
    name: OpenDocument
    description: OpenDocument text document
    icon: ''
    default_app: Collabora
    allow_creation: true
```
This was part of the intense discussion with Willy.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (rhafer) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc:46]: I agree with this. While the detailed diagram is quite interesting, I wonder if we really need that level of detail here. It is also somewhat hard to digest without any further description IMO. I think it would be much better live in the dev docs.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (rhafer) [inline modules/ROOT/pages/deployment/wopi/wopi.adoc:13]: Would it make sense to add a reference to https://en.wikipedia.org/wiki/Web_Application_Open_Platform_Interface or even https://docs.microsoft.com/en-us/openspecs/office_protocols/ms-wopi/6a8bb410-68ad-47e4-9dc3-6cf29c6b046b here to add some background about what WOPI actually is?

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (micbar) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: Still not correct. The app registry config is optional. It works without that. There are two different things:

1) Registering the mimetype -> done via autodiscovery https://onlyoffice.ocis-wopi.latest.owncloud.works/hosting/discovery
2) Providing the actions for the new button and the default app settings. -> app-registry config

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc]: done

File: modules/ROOT/pages/conf-examples/office-integration.adoc
Comment (mmattel) [reply modules/ROOT/pages/conf-examples/office-integration.adoc]: I have chnaged the file names 😄

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: added

**Comment by mmattel** [comment]
Post a discussion with @rhafer, the detailed image (+section) has been removed (see referenced issue to the ocis repo where we can reuse it for dev docs). I also improved some filenames as suggested by @mbarz and improved the text a bit. Based on a suggestion of @rhafer, I added a note in the repos readme on how to access staging. The commits are squashed except the latest.

The current build can be viewed on staging. @rhafer if you are fine, pls approve so we can merge.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: fixed, see my comment below

**Comment by mbarz** [comment]
Please link @micbar instead of me.

**Comment by mmattel** [comment]
> Please link @micbar instead of me.

sure, done, sorry, autocomplete 😢

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc:17]: Also see the [ownCloud Server](https://doc.owncloud.com/server/10.11/admin_manual/enterprise/collaboration/msoffice-wopi-integration.html#introduction) docs:

...If you want to integrate the [Office 365 (cloud)](https://products.office.com/en-us/business/office) version of Microsoft Office Online, you need to [get in touch with us](https://owncloud.com/support)...

And the comment in [Configuring the WOPI App in ownCloud](https://doc.owncloud.com/server/10.11/admin_manual/enterprise/collaboration/msoffice-wopi-integration.html#configuring-the-wopi-app-in-owncloud)

**Review by rhafer — APPROVED** [review]
Looks good to me now. Also from the level of detail, this is ok now IMO.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc:17]: I have an idea how to integrate O365 into this document, but I will file an additional PR.

File: modules/ROOT/pages/deployment/wopi/wopi.adoc
Comment (micbar) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: @mmattel Can you clarify this on a follow up PR?

I think it might be confusing to see more mimetypes attached to apps than provided by the app-registry.yaml

The app-registry.yaml only configures a subset for the "+ New" action. The File open Action and the logo for the file open action are defined otherwise.

Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: sure

Comment (mmattel) [reply modules/ROOT/pages/deployment/wopi/wopi.adoc]: Note that sources like app-registry is/was provided by @wkloucek

**Pull request merged**
